services/movie.py

```diff
@@ -1,40 +1,44 @@
-from django.db.models import QuerySet
+from django.db import transaction
 
 from db.models import Movie
 
 
 def get_movies(
-    genres_ids: list[int] = None,
-    actors_ids: list[int] = None,
-) -> QuerySet:
-    queryset = Movie.objects.all()
+        title: str = None,
+        genres_ids: int = None,
+        actors_ids: int = None
+) -> Movie:
+    movies = Movie.objects.all()
+
+    if title:
+        movies = movies.filter(title__icontains=title)
 
     if genres_ids:
-        queryset = queryset.filter(genres__id__in=genres_ids)
+        movies = movies.filter(genres__id__in=genres_ids).distinct()
 
     if actors_ids:
-        queryset = queryset.filter(actors__id__in=actors_ids)
+        movies = movies.filter(actors__id__in=actors_ids).distinct()
 
```

Comment:
let's simplify it and call distinct at the end because if both ifs are run, then distinct will be called twice that is quite bad because it's an expensive operation

Reply:
didn't fix